utils.py

```diff
@@ -503,6 +504,37 @@ def convert_to_str(string_to_convert):
         return string_to_convert.encode('utf-8')
     return string_to_convert
 
+def get_sorted_nodes(nodes):
+    """Returns the nodes in topological sorted order.
+
+    Args:
+        nodes: list(dict). Unorderd list of nodes.
+
+    Returns:
+        list(dict). The list of nodes in topological sorted order.
+    """
+    unsorted_nodes = list(nodes)
```

Comment:
If it's already a list, why cast it again?

Reply:
As we are deleting the unsorted_nodes values in line 528 and we need a fresh copy of node to return to line 535, So I am just making a copy of nodes to do so.

Reply:
Ah, OK. Use copy.deepcopy() for that sort of thing.